app/planqtn_api/web_endpoints.py

```diff
@@ -56,9 +86,12 @@ async def create_css_tanner_network(request: TannerRequest):
 @router.post("/mspnetwork", response_model=TensorNetworkResponse)
 def create_msp_network(request: TannerRequest):
     try:
+        _validate_matrix_shape(request.matrix)
         matrix = GF2(request.matrix)
         tn = StabilizerMeasurementStatePrepTN(matrix)
         return TensorNetworkResponse.from_tensor_network(tn, request.start_node_index)
+    except HTTPException:
+        raise
     except Exception as e:
         traceback.print_exc()
```

Comment:
I'm okay to have this limit - please add a test to 
https://github.com/planqtn/planqtn/blob/main/app/planqtn_api/css_tanner_network_test.py